source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -109,17 +111,43 @@ Http::FilterHeadersStatus RoleBasedAccessControlFilter::decodeHeaders(Http::Head
                         callbacks_->streamInfo().dynamicMetadata(), nullptr)) {
       ENVOY_LOG(debug, "enforced allowed");
       config_->stats().allowed_.inc();
-      return Http::FilterHeadersStatus::Continue;
     } else {
       ENVOY_LOG(debug, "enforced denied");
       callbacks_->sendLocalReply(Http::Code::Forbidden, "RBAC: access denied", nullptr,
                                  absl::nullopt, RcDetails::get().RbacAccessDenied);
       config_->stats().denied_.inc();
       return Http::FilterHeadersStatus::StopIteration;
     }
+  } else {
+    ENVOY_LOG(debug, "no engine, allowed by default");
+  }
+
+  if (config_->expr() != nullptr) {
```

Comment:
Can we put these logic inside `engine->allowed()`? The `condition` is part of the policy which should also be evaluated by the engine.

Reply:
Done.